File: src/main/java/hzt/aoc/day24/Part1LobbyLayout.java
package hzt.aoc.day24;

import hzt.collections.ListX;
import hzt.collections.MapX;

import java.awt.*;
import java.util.*;
import java.util.List;

public class Part1LobbyLayout extends Day24Challenge {

    public Part1LobbyLayout() {
        super("part 1",
                "Go through the renovation crew's list and determine which tiles they need to flip. " +
                        "After all of the instructions have been followed, how many tiles are left with the black side up?");
    }


    @Override
    protected long calculateResult(List<List<String>> instructionsList) {
        MapX<Point, Tile> tileMap = buildFloorByInstructions(instructionsList);
        return ListX.of(tileMap.values()).count(Tile::isBlackUp);
    }

    @Override
    String getMessage(long result) {
        return String.format("%d", result);
    }

}

File: src/main/java/hzt/aoc/day24/Part1LobbyLayout.kt
package hzt.aoc.day24

class Part1LobbyLayout : Day24Challenge(
    "part 1",
    "Go through the renovation crew's list and determine which tiles they need to flip. " +
            "After all of the instructions have been followed, how many tiles are left with the black side up"
) {
    override fun calculateResult(instructions: List<List<String>>): Long {
        val tileMap = buildFloorByInstructions(instructions)
        return countTilesWithBlackSideUp(tileMap.values)
    }

    override fun getMessage(value: Long): String {
        return String.format("%d", value)
    }
}
